controllers/userController.js

```diff
@@ -1,90 +1,214 @@
-const mysql = require('mysql2');
-
-// Connection Pool
-const connection = mysql.createConnection({
-    host     : process.env.RDS_HOSTNAME,
-    user     : process.env.RDS_USERNAME,
-    password : process.env.RDS_PASSWORD,
-    port     : process.env.RDS_PORT,
-    database : process.env.RDS_DATABASE
-});
-
-connection.connect(
-    function(err) {
-        if (err) throw err;
-        console.log("Connected!");
+const fs = require("fs");
+const { STATUS_CODE, SORT_ORDER, TYPE, REGISTRATION_STATUS } = require('../lib/constants');
+const connection = require('../models/connection');
+
+// Shared functions
+exports.insertUser = ({ email, name, type, visions }) => {
+  const query = `INSERT INTO users (email, name, type, status, visions) VALUES (?,?,?,'unregistered',?)`;
+
+  return new Promise((resolve, reject) => {
+    connection.query(query, [email, name, type, visions], (err, res) => {
+      if (err) reject(err);
+      else resolve(res);
+    })
+  })
+}
+
+exports.verifyUser = ( email ) => {
+  const queryCheck = 'SELECT COUNT(email) AS NUM FROM users WHERE email = ?';
+
+  return new Promise ((resolve, reject) => {
+    connection.query(queryCheck, email, (err, res) => {
+      if (err) reject(err);
+      else resolve(res[0]);
+    })
   });
+}
+
+exports.removeUser = ( email ) => {
+  const query = `DELETE FROM users WHERE email = ?`;
 
-//get all users, return a json object
-exports.viewallusers = async (req, res) => {
-  let message = "view user failed";
-  const query = 'SELECT * FROM users';
-  connection.promise().query(query)
-    .then(data => {
-      message = "view user success";
-      res.status(200).send({message: message, data: data[0]})
+  return new Promise((resolve, reject) => {
+    connection.query(query, email, (err, res) => {
+      if (err) reject(err);
+      else resolve(res);
     })
-    .catch(error => res.status(400).send({message: message, error: error}));
+  });
+}
+
+//load with csv file
+exports.loadfruserLoadfromcsvomcsv = async (req, res) => {
```

Comment:
When something wrong happens during an update, such as repeated email, previous user records are not restored. Perhaps we should first check if the input is valid. If it is, we insert each line into the table. Otherwise, we return the corresponding status code and leave the original table intact.

Reply:
Great point!
Changed it to take in all users that are not duplicates in the file, and the list of duplicated emails will be returned in a list, with status code fo SUCCESS.
When no duplicates are found, status code SUCCESS and empty duplicate array will be returned.